opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/ListTaskPresenter.java

```diff
@@ -460,16 +470,41 @@ private void formatCardDetails(CardDetails cardDetails) {
-        try {
-            String sprayForm = JsonForm.SPRAY_FORM;
+    private void startForm(Feature feature, CardDetails cardDetails, String encounterType) {
+        String taskBusinessStatus = getPropertyValue(feature, TASK_BUSINESS_STATUS);
+        String taskIdentifier = getPropertyValue(feature, TASK_IDENTIFIER);
+        String taskStatus = getPropertyValue(feature, TASK_STATUS);
+
+        String structureId = feature.id();
+        String structureUUID = getPropertyValue(feature, LOCATION_UUID);
+        String structureVersion = getPropertyValue(feature, LOCATION_VERSION);
+        String structureType = getPropertyValue(feature, LOCATION_TYPE);
+        if (SPRAY_EVENT.equals(encounterType)) {
+            String formString = null;
             if (BuildConfig.BUILD_COUNTRY == Country.NAMIBIA) {
-                sprayForm = JsonForm.SPRAY_FORM_NAMIBIA;
+                formString = AssetHandler.readFileFromAssetsFolder(SPRAY_FORM_NAMIBIA, listTaskView.getContext());
             } else if (BuildConfig.BUILD_COUNTRY == Country.BOTSWANA) {
-                sprayForm = JsonForm.SPRAY_FORM_BOTSWANA;
+                formString = AssetHandler.readFileFromAssetsFolder(SPRAY_FORM_BOTSWANA, listTaskView.getContext());
+            } else if (BuildConfig.BUILD_COUNTRY == Country.ZAMBIA) {
+                formString = AssetHandler.readFileFromAssetsFolder(SPRAY_FORM, listTaskView.getContext());
             }
-            String formString = AssetHandler.readFileFromAssetsFolder(sprayForm, listTaskView.getContext());
+            String sprayStatus = cardDetails == null ? null : cardDetails.getSprayStatus();
+            String familyHead = cardDetails == null ? null : cardDetails.getFamilyHead();
+            startForm(formString, structureId, structureUUID, structureVersion, structureType, taskIdentifier,
+                    taskBusinessStatus, taskStatus, structureType, sprayStatus, familyHead);
+        } else if (MOSQUITO_COLLECTION_EVENT.equals(encounterType)) {
+            String formString = null;
+            if (BuildConfig.BUILD_COUNTRY.equals(Country.THAILAND)) {
+                formString = AssetHandler.readFileFromAssetsFolder(THAILAND_MOSQUITO_COLLECTION_FORM, listTaskView.getContext());
+            }
+            startForm(formString, structureId, structureUUID, structureVersion, structureType, taskIdentifier,
+                    taskBusinessStatus, taskStatus, structureType, null, null);
+        }
+    }
+
+    private void startForm(String formString, String structureId, String structureUUID, String structureVersion, String structureType,
+                           String taskIdentifier, String taskBusinessStatus, String taskStatus, String propertyType, String sprayStatus, String familyHead) {
+        try {
             if (StringUtils.isBlank(structureType)) {
                 structureType = StructureType.NON_RESIDENTIAL;
             }
```

Comment:
Remove this because its not applicable to mosquito and larvacide sites

Reply:
Check the data dictionary [here](https://docs.google.com/spreadsheets/d/1kGPgyF0eJdag19FFL1Oog2gdphdPtKobY4EQ0gMgtgw/edit#gid=540134555) and [here](https://docs.google.com/spreadsheets/d/1kGPgyF0eJdag19FFL1Oog2gdphdPtKobY4EQ0gMgtgw/edit#gid=1639363668).

Reply:
This was probably an error, that structure type is only applicable to structures and not mosquito and larval sites. As per the [add location form ](https://docs.google.com/spreadsheets/d/1kGPgyF0eJdag19FFL1Oog2gdphdPtKobY4EQ0gMgtgw/edit#gid=0) when one add a mosquito or larval site it cannot be 
"Residential Structure", "Non-Residential Structure", "Facility"

Reply:
I'm inclined to follow the data dictionary as is.

Maybe you can involve @craigappl to refine this.

Reply:
`structure_type` has been deleted from the data dictionary

Reply:
OK, done.